pkg/infra/kvstore/kvstore_test.go

```diff
@@ -0,0 +1,142 @@
+package kvstore
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+func createTestableKVStore(t *testing.T) *KVStore {
+	t.Helper()
+
+	sqlstore := sqlstore.InitTestDB(t)
+
+	kv := &KVStore{
+		SQLStore: sqlstore,
+	}
+
+	err := kv.Init()
+	require.NoError(t, err)
+
+	return kv
+}
+
+type TestCase struct {
+	OrgId     int64
+	Namespace string
+	Key       string
+	Revision  int64
+}
+
+func (t *TestCase) Value() string {
+	return fmt.Sprintf("%d:%s:%s:%d", t.OrgId, t.Namespace, t.Key, t.Revision)
+}
+
+func TestKVStore(t *testing.T) {
+	kv := createTestableKVStore(t)
+
+	ctx := context.Background()
+
+	testCases := []*TestCase{
+		{
+			OrgId:     0,
+			Namespace: "testing1",
+			Key:       "key1",
+		},
+		{
+			OrgId:     0,
+			Namespace: "testing2",
+			Key:       "key1",
+		},
+		{
+			OrgId:     1,
+			Namespace: "testing1",
+			Key:       "key1",
+		},
+		{
+			OrgId:     1,
+			Namespace: "testing3",
+			Key:       "key1",
+		},
+	}
+
+	for _, tc := range testCases {
+		err := kv.Set(ctx, tc.OrgId, tc.Namespace, tc.Key, tc.Value())
+		require.NoError(t, err)
+	}
+
+	t.Run("get existing keys", func(t *testing.T) {
+		for _, tc := range testCases {
+			value, err := kv.Get(ctx, tc.OrgId, tc.Namespace, tc.Key)
+			require.NoError(t, err)
+			assert.Equal(t, tc.Value(), value)
+		}
+	})
+
+	t.Run("get nonexistent keys", func(t *testing.T) {
+		tcs := []*TestCase{
+			{
+				OrgId:     0,
```

Comment:
In our 1:1 you mentioned that `0` could be global. But I don't see how we pick items with orgId=0 over orgId=1. But perhaps that convention is not needed yet and could just be ignored unless we actually need it.

Reply:
I haven't implemented anything like a fallback to a key in `orgId=0` if it's not found in a specific org, the intent was more that `orgId=0` could be used to store values that are not specific to a particular org.